### PR TITLE
feat: Allow changes to animation duration for marker splitting/merging effects

### LINKED ISSUE #1097: **DefaultClusterRenderer**  -- Add the functionality to change the animation duration for marker splitting/merging effects
In [googlemaps/google-maps-ios-utils](https://github.com/googlemaps/google-maps-ios-utils) we have functionality to change this value by using not private variable [animationDuration](https://github.com/googlemaps/google-maps-ios-utils/blob/main/src/Clustering/View/GMUDefaultClusterRenderer.h#:~:text=%40property(nonatomic)%20double%20animationDuration%3B) into [google-maps-ios-utils/src/Clustering/View/GMUDefaultClusterRenderer.h](https://github.com/googlemaps/google-maps-ios-utils/blob/main/src/Clustering/View/GMUDefaultClusterRenderer.h). It will be great if we have this functionality for android too.

Example of using this variable in iOS:

        guard let mapView = self.mapView else { return }
        let iconGenerator = CustomClusterIconGenerator()
        let renderer = CustomRendererMarkers(mapView: mapView, clusterIconGenerator: iconGenerator)
        renderer.animatesClusters = self.isAnimateClustering
        renderer.animationDuration = self.durationOfClusteringAnimation
        renderer.delegate = self;
        renderer.zIndex = 10;
        // initWithClusterDistancePoints
        let algorithm = GMUNonHierarchicalDistanceBasedAlgorithm(clusterDistancePoints: self.maxDistanceBetweenClusteredItems)
        self.mClusterManager = GMUClusterManager(map: mapView, algorithm: algorithm as! GMUClusterAlgorithm, renderer: renderer)`

Example of possible using this variable in Android:

        OwnIconRenderer ownIconRenderer = new OwnIconRenderer(
                this.context, this.activity, this.googleMap, mClusterManager);
        mClusterManager.setRenderer(ownIconRenderer);
        mClusterManager.getRenderer().setAnimation(this.isAnimateClustering);
        mClusterManager.getRenderer().setAnimationDuration(this.durationOfClusteringAnimation);`

### PR DESCRIPTION
**Pull request by vorokami**
My implementation of my feature request - [issue#1097](https://github.com/googlemaps/android-maps-utils/issues/1097)

Closes #1097

### CODE REVIEW

File: library/src/main/java/com/google/maps/android/clustering/view/ClusterRenderer.java [hunk 1]
```diff
@@ -51,6 +51,11 @@
      */
     void setAnimation(boolean animate);
 
+    /**
+     * Sets the length of the animation. The default duration is 300 milliseconds.
+     */
+    void setAnimationDuration(int animationDuration);
```
Comment (barbeau) [inline library/src/main/java/com/google/maps/android/clustering/view/ClusterRenderer.java:57]: `ValueAnimator.setDuration()` takes in a `long`, so please use a `long` value here and throughout too.

**Review by barbeau — CHANGES_REQUESTED** [review]
Thanks for the PR @vorokami! A few comments inline.

**Comment by vorokami** [comment]
@barbeau. Fixed all comments. Long everywhere for animation duration and Javadocs to interface method and to class method

File: library/src/main/java/com/google/maps/android/clustering/view/DefaultClusterRenderer.java [hunk 2]
```diff
@@ -575,6 +577,15 @@ public void setAnimation(boolean animate) {
         mAnimate = animate;
     }
 
+    /**
+     * {@inheritDoc} The default duration is 300 milliseconds.
+     * @param animationDuration long: The length of the animation, in milliseconds. This value cannot be negative.
+     */
+    @Override
+    public void setAnimationDuration(long animationDuration) {
```
Comment (barbeau) [inline library/src/main/java/com/google/maps/android/clustering/view/DefaultClusterRenderer.java:585]: One last item - can you please change all the variables here to add the `Ms` suffix? For example, `animationDurationMs`, `mAnimationDurationMs`.

Comment (vorokami) [reply library/src/main/java/com/google/maps/android/clustering/view/DefaultClusterRenderer.java]: Done

**Review by barbeau — APPROVED** [review]
Thanks @vorokami!

**Pull request merged**